File: Data Structures & Algorithms/jump-game-ii/submission-0.go
func jump(nums []int) int {
    n := len(nums)
    dp := make([]int, n)
    for i := 1; i < n; i++ {
        dp[i] = -1
    }
    for i := 0; i < n; i++ {
        num := nums[i]
        if dp[i] != -1 {
            for j := i+1; j < n && j <= i+num; j++ {
                if dp[j] == -1 {
                    dp[j] = dp[i] + 1
                } else {
                    dp[j] = min(dp[j], dp[i]+1)
                }
            }
        }
    }
    return dp[n-1]   
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = cur = nums[0]
        cur_min = nums[0]
        for num in nums[1:]:
            if num < 0:
                cur, cur_min = cur_min, cur
            cur = max(num, cur * num)
            cur_min = min(num, cur_min * num)
            res = max(res, cur)
        return res